disnake/flags.py

```diff
@@ -151,6 +132,29 @@ def __init__(self, **kwargs: bool):
                 raise TypeError(f"{key!r} is not a valid flag name.")
             setattr(self, key, value)
 
+    @classmethod
+    def __init_subclass__(cls, inverted: bool = False, no_fill_flags: bool = False):
+        # add a way to bypass filling flags, eg for ListBaseFlags.
+        if no_fill_flags:
+            return cls
+
+        # use the parent's current flags as a base if they exist
+        cls.VALID_FLAGS = getattr(cls, "VALID_FLAGS", {}).copy()
+
+        for name, value in cls.__dict__.items():
+            if isinstance(value, flag_value):
+                value._parent = cls
+                cls.VALID_FLAGS[name] = value.flag
+
+        if not cls.VALID_FLAGS:
+            raise TypeError(
```

Comment:
not sure if this should instead be a `ValueError` or `RuntimeError`

Reply:
Will make it a runtime error.

Reply:
https://github.com/DisnakeDev/disnake/pull/660/commits/016e6ceb79d2a6a650a5b6461139bd7834ab31bf